lib/parameter_substitution.rb

```diff
@@ -69,6 +69,11 @@ def find_tokens(input)
       parse_expression(context).substitution_parameter_names
     end
 
+    def find_formatters(string_with_tokens)
```

Comment:
this method looks very similar to the above `find_tokens` might be good to combing the contracts and maybe dry this up a little bit?